src/raftstore/store/peer.rs

```diff
@@ -626,6 +639,10 @@ impl Peer {
         if self.is_leader() && m.get_from() != INVALID_ID {
             self.peer_heartbeats.insert(m.get_from(), Instant::now());
         }
+        if m.get_from() == self.leader_id() {
```

Comment:
I don't think it works for `Leader`.

Reply:
Do you think https://github.com/pingcap/tikv/pull/3133/commits/a980b7cc794c7bcce592407ab0667839e71a8017 is a good solution?